tests/utils/test_settings_utils.py

```diff
@@ -0,0 +1,164 @@
+        )
+
+        # Check that the directory contains typical Python packages
+        common_packages = ["pip", "django"]
+        found_packages = [
+            pkg
+            for pkg in common_packages
+            if os.path.exists(os.path.join(site_packages_path, pkg))
+        ]
+        self.assertTrue(
+            len(found_packages) > 0,
+            f"Expected {site_packages_path} to contain common Python packages but found none of {common_packages}",
+        )
+
+        # Check that it has .dist-info or .egg-info directories (typical for installed packages)
+        dist_info_files = [
+            f
+            for f in os.listdir(site_packages_path)
+            if f.endswith((".dist-info", ".egg-info"))
+        ]
+        self.assertTrue(
+            len(dist_info_files) > 0,
+            f"Expected {site_packages_path} to contain .dist-info or .egg-info directories",
+        )
+
+        # Check the second call to json.dump has our expected structure
+        expected_tsconfig_data = {
+            "_comment": "This is a generated file. Do not edit directly.",
+            "compilerOptions": {
+                "paths": {
+                    "@/arches/*": [
+                        os.path.join(
+                            ".",
+                            os.path.relpath(
+                                "/arches",
+                                os.path.join("/arches", ".."),
+                            ),
+                            "app",
+                            "src",
+                            "arches",
+                            "*",
+                        )
+                    ],
+                    "@/test_app/*": [
+                        os.path.join(
+                            ".",
+                            os.path.relpath(
+                                "/test/app/path",
+                                os.path.join("/arches", ".."),
+                            ),
+                            "*",
+                        )
+                    ],
+                    "*": ["./node_modules/*"],
+                }
+            },
+        }
+        # Check the second call to json.dump has our expected structure
+        args, _ = mock_json_dump.call_args_list[1]
+        self.assertEqual(
+            set(args[0]["_comment"]), set(expected_tsconfig_data["_comment"])
+        )
+        self.assertEqual(
+            set(args[0]["compilerOptions"]["paths"].keys()),
+            {"@/arches/*", "@/test_app/*", "*"},
+        )
+
+    @mock.patch("arches.settings_utils.sys.stderr.write")
+    def test_generate_frontend_configuration_error_handling(self, mock_stderr_write):
+        """
+        Test error handling in frontend configuration generation.
+        """
+        # Force an exception by removing a required setting
+        with mock.patch("arches.settings_utils.settings") as mock_settings:
```

Comment:
Same as above, you can use `@override_settings` or a variant like `self.modify_settings()`.

Reply:
done